services/src/main/java/org/exoplatform/officeonline/WOPIService.java

```diff
@@ -273,9 +275,13 @@ public String getActionUrl(RequestInfo requestInfo, String fileId, String worksp
     if (title.contains(".")) {
       String extension = title.substring(title.lastIndexOf(".") + 1);
       String actionURL = discoveryPlugin.getActionUrl(extension, action);
-      return String.format("%s%s=%s&", actionURL, PLACEHOLDER_WOPISRC, getWOPISrc(requestInfo, fileId));
+      if (actionURL != null) {
+        return String.format("%s%s=%s&", actionURL, PLACEHOLDER_WOPISRC, getWOPISrc(requestInfo, fileId));
+      } else {
+        throw new ActionNotFoundException("Cannot find actionURL for file extension " + extension + " and action: " + action);
+      }
     } else {
-      throw new OfficeOnlineException("Cannot get file extension. FileId: " + fileId);
+      throw new FileExtensionException("Cannot get file extension. FileId: " + fileId + ". Title: " + title);
```

Comment:
Is it an error for MS Office via WOPI, is it demands a file name with an extension only?
We would guess an extension for a file's mimetype and submit it to their service?

BTW FileExtensionException name doesn't clearly tells about a problem, FileExtensionNotFoundException would be more readable.

Reply:
Renamed to FileExtensionNotFoundException. Commit: https://github.com/exo-addons/office-online/pull/4/commits/855e28198dcd8caf46d27be4e6d513e89c8ea7ed

Reply:
What about guessing an extension by mime-type if it's not found? Will the error, we raise, create a right user experience or better guess an ext and let it just work (rare case)?

Reply:
https://github.com/exo-addons/office-online/pull/4/commits/f8ea42dab3396607852be4cc282229e72b83bef6